python/cudf/cudf/core/frame.py

```diff
@@ -815,6 +815,14 @@ def mask(self, cond, other=None, inplace=False):
 
         return self.where(cond=~cond, other=other, inplace=inplace)
 
+    def _split(self, splits, keep_index=True):
+        result = libcudf.copying.table_split(
+            self, splits, keep_index=keep_index
+        )
+
+        result = [self.__class__._from_table(tbl) for tbl in result]
```

Comment:
Is this making a deep copy of all of the splits?

Reply:
Nope, it does the following:

1. It wraps the table (views) returned by libcudf in an object more familiar to our users (Index/DataFrame/Series/etc.,). This is the `self__class__._from_table(tbl)` bit. This doesn't copy any GPU data.

2. Creates a list of those. This doesn't copy anything either.